testutils/utils.go

```diff
@@ -138,6 +138,7 @@ func SetupTests(m *testing.M) {
 	if finalGoroutineCount != initialGoroutineCount {
 		log.Printf("Detected a goroutine leak (%d before != %d after)", initialGoroutineCount, finalGoroutineCount)
 		pprof.Lookup("goroutine").WriteTo(os.Stdout, 1)
+		result = 1
```

Comment:
We should probably not be overwriting an already non-zero result here?  Unless test.Run() only ever returns 0/1.

Reply:
I believe that tests only return 0/1. 2 Can be returned if testing is interrupted.